flow360/component/simulation/primitives.py

```diff
@@ -395,6 +395,50 @@ class GhostSurface(_SurfaceEntityBase):
     )
 
 
+# pylint: disable=missing-class-docstring
+@final
+class GhostSphere(GhostSurface):
+    private_attribute_entity_type_name: Literal["GhostSphere"] = pd.Field(
+        "GhostSphere", frozen=True
+    )
+    center: List = pd.Field(alias="center")
+    radius: PositiveFloat = pd.Field(alias="radius")
+
+
+# pylint: disable=missing-class-docstring
+@final
+class GhostHemisphere(GhostSurface):
+    private_attribute_entity_type_name: Literal["GhostHemisphere"] = pd.Field(
+        "GhostHemisphere", frozen=True
+    )
+    center: List = pd.Field(alias="center")
+    radius: PositiveFloat = pd.Field(alias="radius")
+    base_normal_axis: List = pd.Field(alias="baseNormalAxis")
+
+
+# pylint: disable=missing-class-docstring
+@final
+class GhostCircularPlane(GhostSurface):
+    private_attribute_entity_type_name: Literal["GhostCircularPlane"] = pd.Field(
+        "GhostCircularPlane", frozen=True
+    )
+    center: List = pd.Field(alias="center")
+    radius: PositiveFloat = pd.Field(alias="radius")
+    normal_axis: List = pd.Field(alias="normalAxis")
+
+
+# pylint: disable=missing-class-docstring
+@final
+class GhostCylindricalCurvedSurface(GhostSurface):
+    private_attribute_entity_type_name: Literal["GhostCylindericalCurvedSurface"] = pd.Field(
+        "GhostCylindericalCurvedSurface", frozen=True
+    )
+    center: List = pd.Field(alias="center")
+    radius: PositiveFloat = pd.Field(alias="radius")
+    height: PositiveFloat = pd.Field(alias="height")
+    axis: List = pd.Field(alias="axis")
```

Comment:
not sure if real dimensions needs to be part of these objects. These are "Ghosts" not real ones.

Reply:
updated